From addcfb42ea8efe81951d8abebd2ba74ce762aad1 Mon Sep 17 00:00:00 2001
From: Fugang Duan <fugang.duan@nxp.com>
Date: Tue, 30 Jan 2018 16:41:17 +0800
Subject: [PATCH 3292/5242] MLK-17475-06 ARM64: dts: imx8qm/qxp: add tja1100
 net card support for mek and arm2 boards

commit  5e0726473fdf0681b320f7c84f82c8154346985b from
https://source.codeaurora.org/external/imx/linux-imx.git

Add NXP TJA1100 net card support for mek and arm2 boards.

For i.MX8QXP MEK/ARM2 board,disable enet1 port due to below issues:
- TJA1100 net card PHY address conflict with enet1 PHY1.
- TJA1100 net card attach on enet2 and use enet2 mdio bus.

For i.MX8QM MEK/ARM2 board, enet1 PHY address shoule be reworked to 2,
otherwise port1 don't work. Of course, the *-tja1100.dtb file focus on
verifying TJA1100 net card.

Reviewed-by: Pandy Gao <pandy.gao@nxp.com>
Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    4 ++
 .../dts/freescale/fsl-imx8qm-enet2-tja1100.dtsi    |   72 ++++++++++++++++++++
 .../fsl-imx8qm-lpddr4-arm2-enet2-tja1100.dts       |   16 +++++
 .../dts/freescale/fsl-imx8qm-mek-enet2-tja1100.dts |   16 +++++
 .../dts/freescale/fsl-imx8qxp-enet2-tja1100.dtsi   |   59 ++++++++++++++++
 .../fsl-imx8qxp-lpddr4-arm2-enet2-tja1100.dts      |   16 +++++
 .../freescale/fsl-imx8qxp-mek-enet2-tja1100.dts    |   16 +++++
 7 files changed, 199 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qm-enet2-tja1100.dtsi
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-enet2-tja1100.dts
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-enet2-tja1100.dts
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qxp-enet2-tja1100.dtsi
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2-enet2-tja1100.dts
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-enet2-tja1100.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 14b1609..0b24ded3 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -23,6 +23,7 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb \
 				 fsl-imx8qm-mek_ca72.dtb \
 				 fsl-imx8qm-mek-hdmi.dtb \
 				 fsl-imx8qm-mek-dsi-rm67191.dtb \
+				 fsl-imx8qm-mek-enet2-tja1100.dtb \
 				 fsl-imx8qm-mek-jdi-wuxga-lvds1-panel.dtb \
 				 fsl-imx8qm-lpddr4-arm2-dp.dtb \
 				 fsl-imx8qm-lpddr4-arm2-8cam.dtb \
@@ -33,13 +34,16 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb \
 				 fsl-imx8qm-lpddr4-arm2-mqs.dtb \
 				 fsl-imx8qm-lpddr4-arm2-usb3.dtb \
 				 fsl-imx8qm-lpddr4-arm2-dsi-rm67191.dtb \
+				 fsl-imx8qm-lpddr4-arm2-enet2-tja1100.dtb \
 				 fsl-imx8qm-lpddr4-arm2-hsic.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb \
 				  fsl-imx8qxp-mek.dtb \
 				  fsl-imx8qxp-mek-dom0.dtb \
 				  fsl-imx8qxp-mek-enet2.dtb \
+				  fsl-imx8qxp-mek-enet2-tja1100.dtb \
 				  fsl-imx8qxp-mek-dsi-rm67191.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-enet2.dtb \
+				  fsl-imx8qxp-lpddr4-arm2-enet2-tja1100.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-gpmi-nand.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-lpspi.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-spdif.dtb \
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-enet2-tja1100.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-enet2-tja1100.dtsi
new file mode 100644
index 0000000..4109542
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-enet2-tja1100.dtsi
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/* fec1 cannot attach to ethphy0 since the PHY address
+ * conflict with ethphy2. So eth0 should not work.
+ * There still enable fec1 to share the MDIO bus for fec2 due
+ * to board limitation.
+ */
+&fec1 {
+	/* PHY address should rework to 2 */
+	phy-handle = <&ethphy2>;
+
+	mdio {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		ethphy0: ethernet-phy@0 {
+			compatible = "ethernet-phy-ieee802.3-c22";
+			reg = <0>;
+			tja110x,refclk_in;
+			/delete-property/ at803x,eee-disabled;
+			/delete-property/ at803x,vddio-1p8v;
+		};
+
+		ethphy2: ethernet-phy@2 {
+			compatible = "ethernet-phy-ieee802.3-c22";
+			reg = <2>;
+			at803x,eee-disabled;
+			at803x,vddio-1p8v;
+		};
+	};
+};
+
+&fec2 {
+	pinctrl-0 = <&pinctrl_fec2_rmii>;
+	clocks = <&clk IMX8QM_ENET1_IPG_CLK>,
+		 <&clk IMX8QM_ENET1_AHB_CLK>,
+		 <&clk IMX8QM_ENET1_REF_50MHZ_CLK>,
+		 <&clk IMX8QM_ENET1_PTP_CLK>,
+		 <&clk IMX8QM_ENET1_TX_CLK>;
+	phy-mode = "rmii";
+	phy-handle = <&ethphy0>;
+	/delete-property/ phy-supply;
+};
+
+&iomuxc {
+	imx8qm-mek {
+		pinctrl_fec2_rmii: fec2rmiigrp {
+			fsl,pins = <
+				SC_P_ENET1_RGMII_TXC_CONN_ENET1_RCLK50M_OUT	0x06000020
+				SC_P_ENET1_RGMII_RXD0_CONN_ENET1_RGMII_RXD0	0x06000020
+				SC_P_ENET1_RGMII_RXD1_CONN_ENET1_RGMII_RXD1	0x06000020
+				SC_P_ENET1_RGMII_RXD2_CONN_ENET1_RMII_RX_ER	0x06000020
+				SC_P_ENET1_RGMII_RX_CTL_CONN_ENET1_RGMII_RX_CTL	0x06000020
+				SC_P_ENET1_RGMII_TXD0_CONN_ENET1_RGMII_TXD0	0x06000020
+				SC_P_ENET1_RGMII_TXD1_CONN_ENET1_RGMII_TXD1	0x06000020
+				SC_P_ENET1_RGMII_TX_CTL_CONN_ENET1_RGMII_TX_CTL	0x06000020
+			>;
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-enet2-tja1100.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-enet2-tja1100.dts
new file mode 100644
index 0000000..42daa8a
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-lpddr4-arm2-enet2-tja1100.dts
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8qm-lpddr4-arm2.dts"
+#include "fsl-imx8qm-enet2-tja1100.dtsi"
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-enet2-tja1100.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-enet2-tja1100.dts
new file mode 100644
index 0000000..817bb62
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-enet2-tja1100.dts
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8qm-mek.dts"
+#include "fsl-imx8qm-enet2-tja1100.dtsi"
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-enet2-tja1100.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-enet2-tja1100.dtsi
new file mode 100644
index 0000000..b664c6f
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-enet2-tja1100.dtsi
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+&fec1 {
+	status = "disabled";
+};
+
+&fec2 {
+	pinctrl-0 = <&pinctrl_fec2_rmii>;
+	clocks = <&clk IMX8QXP_ENET1_IPG_CLK>,
+		 <&clk IMX8QXP_ENET1_AHB_CLK>,
+		 <&clk IMX8QXP_ENET1_REF_50MHZ_CLK>,
+		 <&clk IMX8QXP_ENET1_PTP_CLK>,
+		 <&clk IMX8QXP_ENET1_TX_CLK>;
+	phy-mode = "rmii";
+	phy-handle = <&ethphy2>;
+	/delete-property/ phy-supply;
+
+	mdio {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		ethphy2: ethernet-phy@5 {
+			compatible = "ethernet-phy-ieee802.3-c22";
+			reg = <5>;
+			tja110x,refclk_in;
+		};
+	};
+};
+
+&iomuxc {
+	imx8qxp-mek {
+		pinctrl_fec2_rmii: fec2rmiigrp {
+			fsl,pins = <
+				SC_P_ENET0_MDC_CONN_ENET1_MDC			0x06000020
+				SC_P_ENET0_MDIO_CONN_ENET1_MDIO			0x06000020
+				SC_P_ESAI0_FSR_CONN_ENET1_RCLK50M_OUT		0x06000020
+				SC_P_SPDIF0_RX_CONN_ENET1_RGMII_RXD0            0x06000020
+				SC_P_ESAI0_TX3_RX2_CONN_ENET1_RGMII_RXD1        0x06000020
+				SC_P_ESAI0_TX2_RX3_CONN_ENET1_RMII_RX_ER        0x06000020
+				SC_P_SPDIF0_TX_CONN_ENET1_RGMII_RX_CTL          0x06000020
+				SC_P_ESAI0_TX4_RX1_CONN_ENET1_RGMII_TXD0        0x06000020
+				SC_P_ESAI0_TX5_RX0_CONN_ENET1_RGMII_TXD1        0x06000020
+				SC_P_ESAI0_SCKR_CONN_ENET1_RGMII_TX_CTL         0x06000020
+			>;
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2-enet2-tja1100.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2-enet2-tja1100.dts
new file mode 100644
index 0000000..f45a85a
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2-enet2-tja1100.dts
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8qxp-lpddr4-arm2-enet2.dts"
+#include "fsl-imx8qxp-enet2-tja1100.dtsi"
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-enet2-tja1100.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-enet2-tja1100.dts
new file mode 100644
index 0000000..d9b3842
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-enet2-tja1100.dts
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8qxp-mek-enet2.dts"
+#include "fsl-imx8qxp-enet2-tja1100.dtsi"
-- 
1.7.9.5

